plugins/scaffolder-backend-module-github/src/actions/githubDeployKey.ts

```diff
@@ -152,12 +167,17 @@ export function createGithubDeployKeyAction(options: {
         Sodium.base64_variants.ORIGINAL,
       );
 
-      await client.rest.actions.createOrUpdateRepoSecret({
-        owner: owner,
-        repo: repo,
-        secret_name: privateKeySecretName,
-        encrypted_value: encryptedBase64Secret,
-        key_id: publicKeyResponse.data.key_id,
+      await ctx.checkpoint({
+        key: `create.or.update.repo.secret.${owner}.${repo}.${keyId}`,
```

Comment:
sorry for asking but, how can the secret itself not be part of the key? isn't the `key` part of what can change until the next run?

Reply:
If user will change something, it will be a new task.

We need a unique key here with some extras if for example in a template will be multiple actions, so we won't cache the result of first 'github:deployKey:create' and reuse it for all subsequent 'github:deployKey:create's